### PR TITLE
Update DBIdentityStore fat test with containerauthdata

### LINKED ISSUE #2861: Update DatabaseIdentityStore test with containerAuthData
Update and fix the DatabaseIdentityStore test for containerAuthData. The test uses a containerAuthData to store the user/pwd for the database. Update the test war to have resource ref for the datasource in its web.xml. Update the datasource lookup on the servlet to do an indirect JNDI lookup.

Update the DatabaseIdentityStore to only create one InitialContext object.

### PR DESCRIPTION
**Pull request by kristip17**
fixes #2861

### CODE REVIEW

**Comment by kristip17** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.javaeesec_fat

**Comment by kristip17** [comment]
#libby

**Comment by kristip17** [comment]
For the build, javaeesec fat tests all ran successfully. There was a timeout issue overnight again, which looks like it caused some other unrelated test failures.

File: dev/com.ibm.ws.security.javaeesec_fat/test-applications/dbfatAuthAlias.war/src/web/war/db/CustomQueryDatabaseServlet.java [hunk 1]
```diff
@@ -24,40 +24,55 @@
...
+                stmt1.close();
 
+            } finally {
+                conn.close();
+            }
         } catch (Exception e) {
             System.out.println("Failed to create database for DatabaseIdentityStore: " + e.getMessage());
             e.printStackTrace();
+        } finally {
+
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.javaeesec_fat/test-applications/dbfatAuthAlias.war/src/web/war/db/CustomQueryDatabaseServlet.java]: Did we intend to have an empty finally block?

Comment (kristip17) [reply dev/com.ibm.ws.security.javaeesec_fat/test-applications/dbfatAuthAlias.war/src/web/war/db/CustomQueryDatabaseServlet.java]: Removed empty finally block.

**Pull request merged**